asset-registry/src/lib.rs

```diff
@@ -0,0 +1,242 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::pallet_prelude::*;
+use frame_support::traits::EnsureOrigin;
+use frame_support::transactional;
+use frame_system::pallet_prelude::*;
+pub use module::*;
+use orml_traits::asset_registry::AssetProcessor;
+use scale_info::TypeInfo;
+use sp_runtime::traits::AtLeast32BitUnsigned;
+use sp_runtime::{traits::Member, DispatchResult};
+use sp_std::prelude::*;
+use xcm::v2::prelude::*;
+use xcm::VersionedMultiLocation;
+
+pub use impls::*;
+pub use weights::WeightInfo;
+
+mod impls;
+mod mock;
+mod tests;
+mod weights;
+
+#[derive(scale_info::TypeInfo, Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug)]
+pub struct AssetMetadata<Balance, CustomMetadata: Parameter + Member + TypeInfo> {
+	pub decimals: u32,
+	pub name: Vec<u8>,
+	pub symbol: Vec<u8>,
+	pub existential_deposit: Balance,
+	pub location: Option<VersionedMultiLocation>,
+	pub additional: CustomMetadata,
+}
+
+#[frame_support::pallet]
+pub mod module {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Additional non-standard metadata to store for each asset
+		type CustomMetadata: Parameter + Member + TypeInfo;
+
+		/// The type used as a unique asset id,
+		type AssetId: Parameter + Member + TypeInfo;
+
+		/// The origin that is allowed to manipulate metadata.
+		type AuthorityOrigin: EnsureOrigin<<Self as frame_system::Config>::Origin>;
+
+		/// A filter ran upon metadata registration that assigns an is and
+		/// potentially modifies the supplied metadata.
+		type OnRegisterAsset: AssetProcessor<Self::AssetId, AssetMetadata<Self::Balance, Self::CustomMetadata>>;
+
+		/// The balance type.
+		type Balance: Parameter
+			+ Member
+			+ AtLeast32BitUnsigned
+			+ Default
+			+ Copy
+			+ MaybeSerializeDeserialize
+			+ Into<u128>;
+
+		// /// Weight information for extrinsics in this module.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Asset was not found
+		AssetNotFound,
+		/// The version of the `Versioned` value used is not able to be
+		/// interpreted.
+		BadVersion,
+		/// The asset id is invalid.
+		InvalidAssetId,
+		/// Another asset was already register with this location.
+		ConflictingLocation,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		RegisteredAsset {
+			asset_id: T::AssetId,
+			metadata: AssetMetadata<T::Balance, T::CustomMetadata>,
+		},
+		UpdatedAsset {
+			asset_id: T::AssetId,
+			metadata: AssetMetadata<T::Balance, T::CustomMetadata>,
+		},
+		SetLocation {
+			asset_id: T::AssetId,
+			location: Box<VersionedMultiLocation>,
+		},
+	}
+
+	/// The total issuance of a token type.
+	#[pallet::storage]
+	#[pallet::getter(fn get_metadata)]
```

Comment:
```suggestion
	#[pallet::getter(fn metadata)]
```

no `get_` in getter
https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter